### PR TITLE
[#1318] PrepWork 2+3/3: Added cancellable ability to util/poll.js

### PR DESCRIPTION
**Pull request by ranbena**
Preparation work for solving #1318.

**tl;dr** Added cancellablePoll

NOTICE: This PR is dependent on #1730. Please defer review until merged.

#### The #1318 Pull Request stack
1. (#1730) PrepWork 1/3: Added util/poll.js unit tests
2. (#1731) PrepWork 2/3: Made util/poll.js cancellable with cancellablePoll
3. (#1731) PrepWork 3/3: Added cancellablePoll unit tests ☜☜☜ _**YOU ARE HERE**_
4. (#1732) Actual 1/2: Enable “Scan QR” button only if camera available
5. (#1733) Actual 2/2: Added camera availability unit tests

#### What's this commit about?
This is prepare work for the actual feature - Enabling/disabling “Scan QR” button depending on camera availability.
The main feature (next commit) polls continuously for camera availability. It's important to cancel the polling if the user clicks away from the login tab.

**Tests:**
Previous poll tests keep passing.
Added new "should be cancellable" test

- [ ] Unit tests written?

### CODE REVIEW

File: app/util/poll.js [hunk 1]
```diff
@@ -3,17 +3,47 @@ import delay from './delay'
...
+  }
+}
+
+export function cancellablePoll(
   request,
   { attempts = DEFAULT_ATTEMPTS, frequency = DEFAULT_FREQUENCY } = {}
 ) {
-  return request().catch(function retry(err) {
+  let hasCancelled = false
+  const promise = request().catch(function retry() {
```
Comment (comountainclimber) [inline app/util/poll.js]: Maybe Im missing something but wouldn't it be better to define `retry` outside of this catch block so that it doesn't get redefined every time `request` throws an error?

Comment (ranbena) [reply app/util/poll.js]: Indeed. I'll change, test and PR separately.

Comment (ranbena) [reply app/util/poll.js]: Pull request #1734
